app/models/annotation.rb

```diff
@@ -172,4 +172,32 @@ def touch_resource
     true
   end
 
+  def self.reassess_annotations_for_taxon_ids( taxon_ids )
+    Annotation.
+        joins(
+          controlled_attribute: {
+            controlled_term_taxa: {
+              taxon: :taxon_ancestors
+            }
+          }
+        ).
+        where( "taxon_ancestors.ancestor_taxon_id IN (?)", taxon_ids ).
+        includes(
```

Comment:
I was looking at the queries this would generate if run on a large taxon like Animalia. Looks like we can reduce the number of queries needed for validation with a little more preloading. This worked for me locally:

```
        includes(
          { resource: :taxon },
          controlled_value: [ :taxa, :excepted_taxa, {
            controlled_term_taxa: :taxon
          } ],
          controlled_attribute: [ :values, :taxa, :excepted_taxa, {
            controlled_term_taxa: :taxon
          } ]
        ).
```